### PR TITLE
[Enhancement] Add type annoations for mmcv/visualization/image

### PR DESCRIPTION
**Pull request by sundongwei**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

I wanna become a contributor for MMCV, so I try to PR some easy tasks for a practice.

## Modification

I added some Type Hints in image.py which path is "mmcv/mmcv/visualization".

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/visualization/image.py [hunk 1]
```diff
@@ -27,15 +31,15 @@ def imshow(img, win_name='', wait_time=0):
...
-                  bboxes,
-                  colors='green',
-                  top_k=-1,
-                  thickness=1,
-                  show=True,
-                  win_name='',
-                  wait_time=0,
-                  out_file=None):
+def imshow_bboxes(img: Union[str, np.ndarray],
+                  bboxes: List[int, float],
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:35]:
```suggestion
                  bboxes: Union[int, float],
```

**Comment by zhouzaida** [comment]
Please update your branch by merging mmcv master branch.

**Comment by zhouzaida** [comment]
Please update your branch by merging mmcv master branch.

File: mmcv/visualization/image.py [hunk 2]
```diff
@@ -1,12 +1,16 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional, Union
+
 import cv2
 import numpy as np
 
 from mmcv.image import imread, imwrite
 from .color import color_val
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:8]:
```suggestion
from .color import color_val, Color
```

File: mmcv/visualization/image.py [hunk 3]
```diff
@@ -27,15 +31,15 @@ def imshow(img, win_name='', wait_time=0):
...
-                  colors='green',
-                  top_k=-1,
-                  thickness=1,
-                  show=True,
-                  win_name='',
-                  wait_time=0,
-                  out_file=None):
+def imshow_bboxes(img: Union[str, np.ndarray],
+                  bboxes: Union[list, np.ndarray],
+                  colors: Union[str, tuple, int, np.ndarray] = 'green',
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:36]:
```suggestion
                  colors: Union[Color, str, tuple, int, np.ndarray] = 'green',
```

File: mmcv/visualization/image.py [hunk 4]
```diff
@@ -81,19 +85,19 @@ def imshow_bboxes(img,
...
-                      show=True,
-                      win_name='',
-                      wait_time=0,
-                      out_file=None):
+def imshow_det_bboxes(img: Union[str, np.ndarray],
+                      bboxes: np.ndarray,
+                      labels: np.ndarray,
+                      class_names: List[str] = None,
+                      score_thr: float = 0,
+                      bbox_color: Union[str, tuple] = 'green',
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:93]:
```suggestion
                      bbox_color: Union[str, tuple, int, np.ndarray] = 'green',
```

File: mmcv/visualization/image.py [hunk 5]
```diff
@@ -81,19 +85,19 @@ def imshow_bboxes(img,
...
+                      text_color: Union[str, tuple] = 'green',
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:94]:
```suggestion
                      text_color: Union[str, tuple, int, np.ndarray] = 'green',
```

Comment (zhouzaida) [reply mmcv/visualization/image.py:94]: Please help update the type in https://github.com/open-mmlab/mmcv/pull/1946/files#diff-7ab945ee4fc4c1df5be4905c547ba7fb381e381a04876b1968444b4689c2df9bR111

File: mmcv/visualization/image.py (hunk 3 above)
Comment (zhouzaida) [reply mmcv/visualization/image.py:36]: Please help update the type in https://github.com/open-mmlab/mmcv/pull/1946/files#diff-7ab945ee4fc4c1df5be4905c547ba7fb381e381a04876b1968444b4689c2df9bR48

File: mmcv/visualization/image.py [hunk 6]
```diff
@@ -103,8 +109,8 @@ def imshow_det_bboxes(img,
         labels (ndarray): Labels of bboxes.
         class_names (list[str]): Names of each classes.
         score_thr (float): Minimum score of bboxes to be shown.
-        bbox_color (str or tuple or :obj:`Color`): Color of bbox lines.
-        text_color (str or tuple or :obj:`Color`): Color of texts.
+        bbox_color (Color, str, tuple, int, ndarray): Color of bbox lines.
+        text_color (Color, str, tuple, int, ndarray): Color of texts.
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:113]:
```suggestion
        bbox_color (Color or str or tuple or int or ndarray): Color of bbox lines.
        text_color (Color or str or tuple or int or ndarray): Color of texts.
```

File: mmcv/visualization/image.py [hunk 7]
```diff
@@ -1,12 +1,16 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional, Union
+
 import cv2
 import numpy as np
 
 from mmcv.image import imread, imwrite
-from .color import color_val
+from .color import Color, color_val
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:8]:
```suggestion
from .color import Color, color_val

ColorType = Union[Color, str, tuple, int, np.ndarray]
```

File: mmcv/visualization/image.py [hunk 8]
```diff
@@ -27,21 +31,21 @@ def imshow(img, win_name='', wait_time=0):
...
-                  colors='green',
-                  top_k=-1,
-                  thickness=1,
-                  show=True,
-                  win_name='',
-                  wait_time=0,
-                  out_file=None):
+def imshow_bboxes(img: Union[str, np.ndarray],
+                  bboxes: Union[list, np.ndarray],
+                  colors: Union[Color, str, tuple, int, np.ndarray] = 'green',
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:36]:
```suggestion
                  colors: ColorType = 'green',
```

File: mmcv/visualization/image.py [hunk 9]
```diff
@@ -81,19 +85,21 @@ def imshow_bboxes(img,
...
-                      out_file=None):
+def imshow_det_bboxes(img: Union[str, np.ndarray],
+                      bboxes: np.ndarray,
+                      labels: np.ndarray,
+                      class_names: List[str] = None,
+                      score_thr: float = 0,
+                      bbox_color: Union[Color, str, tuple, int,
+                                        np.ndarray] = 'green',
+                      text_color: Union[Color, str, tuple, int,
+                                        np.ndarray] = 'green',
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:96]:
```suggestion
                      bbox_color: ColorType = 'green',
                      text_color: ColorType = 'green',
```

**Review by zhouzaida — APPROVED** [review]

File: mmcv/visualization/image.py [hunk 10]
```diff
@@ -1,12 +1,21 @@
...
 import numpy as np
 
 from mmcv.image import imread, imwrite
-from .color import color_val
+from .color import Color, color_val
+
+ColorType = Union[Color, str, tuple, int, np.ndarray]
+"""
+Define Args colors, bbox_color and text_color Type.
+"""
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:13]:
```suggestion
# a type alias declares the optional types of color argument
ColorType = Union[Color, str, tuple, int, np.ndarray]
```

File: mmcv/visualization/image.py [hunk 11]
```diff
@@ -1,12 +1,18 @@
 # Copyright (c) OpenMMLab. All rights reserved.
+from typing import List, Optional, Union
+
 import cv2
 import numpy as np
 
 from mmcv.image import imread, imwrite
-from .color import color_val
+from .color import Color, color_val
+# a type alias declares the optional types of color argument
```
Comment (zhouzaida) [inline mmcv/visualization/image.py:9]:
```suggestion

# a type alias declares the optional types of color argument
```

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @sundongwei ！First of all, we want to express our gratitude for your significant PR in the mmcv project. Your contribution is highly appreciated, and we are grateful for your efforts in helping improve this open-source project during your personal time. We believe that many developers will benefit from your PR.

We would also like to invite you to join our Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/raweFPmdzG

If you have WeChat，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request merged**